CMakeLists.txt

```diff
@@ -1,6 +1,76 @@
-# SPDX-License-Identifier: Apache-2.0
-#
+# Copyright (c) 2022 Cypress Semiconductor Corporation.
 # Copyright (c) 2020 Linumiz
-# Author: Parthiban Nallathambi <parthiban@linumiz.com>
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_compile_definitions(${CONFIG_SOC_PART_NUMBER})
 
+## Add PDL sources for XMC devices
 add_subdirectory_ifdef(CONFIG_HAS_XMCLIB XMCLib)
+
+## Add PDL, HAL, and other sources for CAT1A devices
+if(CONFIG_SOC_FAMILY_CAT1)
+  # Add abstraction-rtos
+  zephyr_include_directories(abstraction-rtos/include/)
+
+  # CMSIS_RTOS_V2 support
+  zephyr_include_directories_ifdef(CONFIG_CMSIS_RTOS_V2 abstraction-rtos/include/COMPONENT_RTX)
+  zephyr_library_sources_ifdef(CONFIG_CMSIS_RTOS_V2 abstraction-rtos/source/COMPONENT_RTX/cyabs_rtos_rtxv5.c)
+
+  # Zephyr support
+  zephyr_include_directories_ifdef(COMPONENT_ZEPHYR abstraction-rtos/include/COMPONENT_ZEPHYR)
+  zephyr_library_sources_ifdef(COMPONENT_ZEPHYR abstraction-rtos/source/COMPONENT_ZEPHYR/cyabs_rtos_zephyr.c)
+endif()
+
+if(CONFIG_SOC_FAMILY_CAT1A)
+  # Add mtb-hal-cat1
+  zephyr_include_directories(mtb-hal-cat1/include/)
+  zephyr_include_directories(mtb-hal-cat1/COMPONENT_PSOC6HAL/include/)
+  zephyr_include_directories(mtb-hal-cat1/COMPONENT_PSOC6HAL/COMPONENT_CAT1A/include/)
+
+  FILE(GLOB hal_sources mtb-hal-cat1/COMPONENT_PSOC6HAL/source/*.c)
+  target_sources(app PRIVATE ${hal_sources})
+
+  FILE(GLOB hal_sources mtb-hal-cat1/COMPONENT_PSOC6HAL/COMPONENT_CAT1A/source/pin_packages/*.c)
+  target_sources(app PRIVATE ${hal_sources})
```

Comment:
why is this added to the `app` library ?

Reply:
updated